### ISSUE TITLE
Parsing models where the key is also the root of the next node

### ISSUE THREAD

**Issue by gus-byrnesinnovation** [Mantle/Mantle#575]
Hello I have Json that looks like this:
 
    { 
      "items":{
        "devices.hasipad":{"value":false},
        "isConnected":{"value":false}}
    }

The key for the inner json is the "name" attribute on the Item model with a value of "devices.hasipad"

It in turn has a value of false. Idealy I would like to store it in a model with attributes like so:

  @property NSString *name
  @property Boolean isOn

Where in the json above would be name =  "devices.hasipad" and isOn = false

What would be the best approach to take to parse this?

**Comment by robb** [comment]
You could write a custom value transformer for the items property that does the transformation.

**Comment by gus-byrnesinnovation** [comment]
Thanks @robb .

The custom value transformer makes sense. I did not realize I had the option of specifying a custom transformer for a class. I was trying to figure out how to do a custom transformer on the attribute level which was throwing me off since I could not figure out the mapping for a dynamic.

Found this doc for anyone coming across this on a google search which has a nice example:

http://www.fantageek.com/1699/understanding-mantle-value-transformer/
